src/Common/tests/Cryptography/AlgorithmImplementations/ECDsa/ECDsaTests.cs

```diff
@@ -0,0 +1,473 @@
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataByteArraySpan_OffsetGreaterThanCount_ThrowsArgumentOutOfRangeException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("offset",
+                () => ecdsa.VerifyData(new byte[0], 2, 1, null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataByteArraySpan_NegativeCount_ThrowsArgumentOutOfRangeException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("count",
+                () => ecdsa.VerifyData(new byte[0], 0, -1, null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataByteArraySpan_CountGreaterThanLengthMinusOffset_ThrowsArgumentOutOfRangeException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("count",
+                () => ecdsa.VerifyData(new byte[0], 0, 1, null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataByteArraySpan_NullSignature_ThrowsArgumentNullException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentNullException>("signature",
+                () => ecdsa.VerifyData(new byte[0], 0, 0, null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataByteArraySpan_EmptyHashAlgorithm_ThrowsArgumentException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentException>("hashAlgorithm",
+                () => ecdsa.VerifyData(new byte[10], 0, 10, new byte[0], new HashAlgorithmName("")));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataStream_NullData_ThrowsArgumentNullException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentNullException>("data",
+                () => ecdsa.VerifyData((Stream)null, null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataStream_NullSignature_ThrowsArgumentNullException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentNullException>("signature",
+                () => ecdsa.VerifyData(new MemoryStream(), null, default(HashAlgorithmName)));
+        }
+
+        [Theory, MemberData("AllImplementations")]
+        public void VerifyDataStream_DefaultHashAlgorithm_ThrowsArgumentException(ECDsa ecdsa)
+        {
+            Assert.Throws<ArgumentException>("hashAlgorithm",
+                () => ecdsa.VerifyData(new MemoryStream(), new byte[0], default(HashAlgorithmName)));
+        }
+
+        //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+        [Fact]
+        public void SignDataByteArray_BaseWithValidArguments_ThrowsNotImplementedBecauseHashDataOverrideIsNeeded()
```

Comment:
This chunk of tests can be removed; they exist in the desktop tests because the HashData overload with HashAlgorithmName had to be implemented as virtual-throw instead of abstract for compatibility reasons.  It doesn't make as much sense here where we were able to call the methods abstract.